### PR TITLE
reduce accessor types

### PR DESCRIPTION
**Pull request by mtao**
removed const and mutable accessor types

### CODE REVIEW

File: src/wmtk/attribute/Accessor.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "TupleAccessor.hpp"
+
+#include <Eigen/Dense>
+
+namespace wmtk::attribute {
+
+template <typename T>
+class Accessor: public TupleAccessor<T> {
```
Comment (daniel-zint) [inline src/wmtk/attribute/Accessor.hpp:10]: `Accessor` seems to be redundant. Maybe just replace it by `TupleAccessor`?

File: src/wmtk/attribute/MutableAccessor.hpp [hunk 2]
```diff
@@ -1,67 +1,8 @@
...
-    using ConstAccessorType::ConstAccessorType;
-
-    using ConstAccessorType::const_scalar_attribute;
-    using ConstAccessorType::const_vector_attribute;
-    using ConstAccessorType::dimension;
-    using ConstAccessorType::reserved_size;
-    using ConstAccessorType::scalar_attribute;
-    using ConstAccessorType::vector_attribute;
-
+using MutableAccessor = TupleAccessor<T>;
```
Comment (daniel-zint) [inline src/wmtk/attribute/MutableAccessor.hpp:6]: I think `MutableAccessor` can be removed completely.

File: src/wmtk/attribute/TupleAccessor.hpp [hunk 3]
```diff
@@ -19,6 +25,9 @@ class TupleAccessor : protected CachingAccessor<T>
     friend class wmtk::Mesh;
     friend class wmtk::TetMesh;
     friend class wmtk::TriMesh;
+    friend class wmtk::EdgeMesh;
+    friend class wmtk::PointMesh;
+    friend class wmtk::TriMeshOperationExecutor;
```
Comment (daniel-zint) [inline src/wmtk/attribute/TupleAccessor.hpp:30]: It's a bit odd that `TriMeshOperationExecutor` needs to be a friend but `TetMeshOperationExecutor` does not.

File: src/wmtk/attribute/TupleAccessor.hpp [hunk 4]
```diff
@@ -30,10 +39,10 @@ class TupleAccessor : protected CachingAccessor<T>
...
     using CachingBaseType::CachingBaseType;
 
-    TupleAccessor(TupleAccessor&&) = default;
-    TupleAccessor& operator=(TupleAccessor&&) = default;
-    TupleAccessor(const TupleAccessor&) = delete;
-    TupleAccessor& operator=(const TupleAccessor&) = delete;
+    //TupleAccessor(TupleAccessor&&) = default;
+    //TupleAccessor& operator=(TupleAccessor&&) = default;
+    //TupleAccessor(const TupleAccessor&) = delete;
+    //TupleAccessor& operator=(const TupleAccessor&) = delete;
```
Comment (daniel-zint) [inline src/wmtk/attribute/TupleAccessor.hpp:45]: delete?

**Review by daniel-zint — COMMENTED** [review]
Looks good but I think there is some clean-up required still.

File: src/wmtk/attribute/MutableAccessor.hpp (hunk 2 above)
Comment (mtao) [reply src/wmtk/attribute/MutableAccessor.hpp:6]: i thought i'd deleted this file :upside_down_face: . wilco

File: src/wmtk/attribute/Accessor.hpp (hunk 1 above)
Comment (mtao) [reply src/wmtk/attribute/Accessor.hpp:10]: i'm a bit torn on this - i like having the explicit semantic that a TupleAccessor is an Accessor. I guess i'll just replace TupleAccessor with Accessor (i don't want users to think about the Tuple aspect too much)

File: src/wmtk/attribute/TupleAccessor.hpp (hunk 3 above)
Comment (mtao) [reply src/wmtk/attribute/TupleAccessor.hpp:30]: yeah, i hadn't thought about that, but for whatever reason this was required

File: src/wmtk/attribute/TupleAccessor.hpp (hunk 4 above)
Comment (mtao) [reply src/wmtk/attribute/TupleAccessor.hpp:45]: yep. forwarding of constructors wasn't working properly and i was testing.

**Pull request merged**